libass/ass_outline.h

```diff
@@ -38,20 +38,31 @@ typedef struct
     int32_t x_min, y_min, x_max, y_max;
 } ASS_Rect;
 
-typedef struct ass_outline {
-    size_t n_contours, max_contours;
-    size_t *contours;
+enum {
+    OUTLINE_LINE_SEGMENT     = 1,
+    OUTLINE_QUADRATIC_SPLINE = 2,
+    OUTLINE_CUBIC_SPLINE     = 3,
+    OUTLINE_COUNT_MASK       = 3,
+    OUTLINE_CONTOUR_END      = 4
```

Comment:
It would perhaps be useful to add a comment here to explain what the values of these constants mean. As far as I can tell:

* `OUTLINE_LINE_SEGMENT`, `OUTLINE_QUADRATIC_SPLINE` and `OUTLINE_CUBIC_SPLINE` show how many points (minus the starting point) define the respective segment/spline;
* a `segment` value actually consists of two bit fields: one is the number of points as above, and it can be extracted using `OUTLINE_COUNT_MASK`; the other is a single-bit flag showing whether this is the last segment of a contour, `OUTLINE_CONTOUR_END`.

I’m not sure whether it’s worth noting that this enum must not contain zero values as zero is used to mean “don’t add any segment” in `outline_add_point` calls, or whether it’s worth making that explicit by giving zero a name and using it in those calls.

Reply:
I dislike having named constant for zero here due to zero being prohibited value in `segments`. I guess I mention zero in comment before `outline_add_point()`.